types/address.go

```diff
@@ -145,73 +108,39 @@ var (
 // ----------------------------------------------------------------------------
 
 // AccAddress a wrapper around bytes meant to represent an account address.
-// When marshaled to a string or JSON, it uses Bech32.
+// When marshaled to a string or JSON, it uses hex.
 type AccAddress []byte
 
-// AccAddressFromHexUnsafe creates an AccAddress from a HEX-encoded string.
-//
-// Note, this function is considered unsafe as it may produce an AccAddress from
-// otherwise invalid input, such as a transaction hash. Please use
-// AccAddressFromBech32.
-func AccAddressFromHexUnsafe(address string) (addr AccAddress, err error) {
+// AccAddressFromHex creates an AccAddress from a HEX-encoded string.
+func AccAddressFromHex(address string) (addr AccAddress, err error) {
 	bz, err := addressBytesFromHexString(address)
 	return AccAddress(bz), err
 }
 
 // VerifyAddressFormat verifies that the provided bytes form a valid address
-// according to the default address rules or a custom address verifier set by
-// GetConfig().SetAddressVerifier().
-// TODO make an issue to get rid of global Config
-// ref: https://github.com/cosmos/cosmos-sdk/issues/9690
 func VerifyAddressFormat(bz []byte) error {
-	verifier := GetConfig().GetAddressVerifier()
-	if verifier != nil {
-		return verifier(bz)
-	}
-
 	if len(bz) == 0 {
 		return errorsmod.Wrap(sdkerrors.ErrUnknownAddress, "addresses cannot be empty")
 	}
 
-	if len(bz) > address.MaxAddrLen {
-		return errorsmod.Wrapf(sdkerrors.ErrUnknownAddress, "address max length is %d, got %d", address.MaxAddrLen, len(bz))
+	if !common.IsHexAddress(common.BytesToAddress(bz).String()) {
```

Comment:
I'm wondering if we should call`common.BytesToAddress(bz)` since
`common.BytesToAddress` prefixes `0..000` to make it 20 byte (see [this](https://go.dev/play/p/mnGJCKjJb29)). The check would pass for  invalid address strings like "foo", "bar", etc.

Reply:
Changed to
`if !common.IsHexAddress(common.Bytes2Hex(bz)) {`
where "padding" does not apply
Will push

Reply:
Pushed 🙌